paasta_tools/kubernetes_tools.py

```diff
@@ -1416,22 +1441,62 @@ def get_boto_volume(self) -> Optional[V1Volume]:
                     path=this_key,
                 )
                 items.append(item)
-        # Check that boto keys actually exist as secrets
+        # Assume k8s secret exists if its configmap signature exists
         secret_hash = self.get_boto_secret_hash()
         if not secret_hash:
             log.warning(f"Expected to find k8s secret {secret_name} for boto_cfg")
             return None
-        secret_name = limit_size_with_hash(f"paasta-boto-key-{service_name}")
+
         volume = V1Volume(
             name=self.get_boto_secret_volume_name(service_name),
             secret=V1SecretVolumeSource(
-                secret_name=secret_name,
+                secret_name=self.get_boto_k8s_secret_name(),
                 default_mode=mode_to_int("0444"),
                 items=items,
             ),
         )
         return volume
 
+    def get_crypto_keys_from_config(self) -> List[str]:
+        crypto_keys = self.config_dict.get("crypto_keys", {})
+        return [
+            *(f"public/{key}" for key in crypto_keys.get("encrypt", [])),
+            *(f"private/{key}" for key in crypto_keys.get("decrypt", [])),
+        ]
+
+    def get_crypto_volume(self) -> Optional[V1Volume]:
+        required_crypto_keys = self.get_crypto_keys_from_config()
+        if not required_crypto_keys:
+            return None
+
+        if not self.get_crypto_secret_hash():
+            log.warning(
+                f"Expected to find secret signature {self.get_crypto_k8s_secret_name()} for crypto_keys"
+            )
+            return None
+
+        return V1Volume(
+            name=self.get_crypto_secret_volume_name(
+                self.get_sanitised_deployment_name()
+            ),
+            secret=V1SecretVolumeSource(
+                secret_name=self.get_crypto_k8s_secret_name(),
+                default_mode=mode_to_int("0444"),
+                items=[
+                    V1KeyToPath(
+                        # key should exist in data section of k8s secret
+                        key=get_vault_key_secret_name(crypto_key),
+                        # path is equivalent to Vault key directory structure
+                        # e.g. private/foo will create /etc/crypto_keys/private/foo.json
+                        path=f"{crypto_key}.json",
+                        mode=mode_to_int("0444"),
+                    )
+                    for crypto_key in required_crypto_keys
+                ],
+                optional=True,
```

Comment:
just to confirm: we're expecting that users of this feature check that these secrets are available before using them, right? otherwise, we'd want to remove this optional flag so that we don't run the pod if we cannot use the volume